docs/development_guide.md

```diff
@@ -0,0 +1,121 @@
+Development Guide
+-----------------
+
+This guide is for those that want to help hack on the Slack bridge, and need some instruction on how
+to set it up. This is **not** a guide for setting up an instance for production usage.
+
+For information on how to contribute issues and or pull requests, please read [CONTRIBUTING](../CONTRIBUTING.md).
+
+## Setting up your environment
+
+This section explains how to setup Synapse, Riot and the Slack bridge for local development.
+
+### Prereqs 
+
+Ensure at the minimum you have installed:
+
+- NodeJS ( [nvm](https://github.com/nvm-sh/nvm) is a good tool for this)
+- Docker (optional, but it will make your life easier)
+
+For the sake of making this easier to follow, we will create a directory in the home directory
+called `slack-bridge-env`:
+
+```bash
+mkdir ~/slack-bridge-env
+cd ~/slack-bridge-env
+```
+
+### Setting up Synapse
+
+Largely to setup Synapse, you can follow https://hub.docker.com/r/matrixdotorg/synapse and just ensure your data directory points to `~/slack-bridge-env/synapse`.  
+You will want to make sure your server name is something that routes to your local box. I tend to use the devbox hostname, but `localhost` is also sufficent.  
+You should enable registration in the `homeserver.yaml` to create a testing user.  
```

Comment:
Can be explicit:

Find and uncomments the `enable_registration` field, set it to `true`:

```
enable_registration: true
```